src/query/models/tags.go

```diff
@@ -279,8 +279,8 @@ func (t Tags) Validate() error {
 
 			prev := tags.Tags[i-1]
 			if bytes.Compare(prev.Name, tag.Name) == 0 {
-				return fmt.Errorf("tags duplicate: '%s' appears more than once",
-					tags.Tags[i-1].Name)
+				return fmt.Errorf("tags duplicate: '%s' appears more than once in '%s'",
+					tags.Tags[i-1].Name, tags.id)
```

Comment:
As previously mentioned, not guaranteed that `t.id` is built at this point. 

Perhaps best to use dynamically built string output, e.g. `t.String()` which uses a string builder.

Reply:
Or perhaps call `quotedID(...)` directly to ensure a quoted version is always printed (rather than graphite if this is a graphite set of tags, etc).

Reply:
I think you can even omit the `.String()` since it's a `io.Stringer`, 

i.e.
`return fmt.Errorf("tags duplicate: '%s' appears more than once in '%s'", tags.Tags[i-1].Name, tags)`